From fb8ad71872a2aeb20cb967cd88a0a3577d49dbca Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Tue, 30 Apr 2024 08:35:43 +0530
Subject: [PATCH] Fixed for Coverity issue in Car Service.

Captured concurrent data modification issue.

Synchronized data modification variable.

Tests: Build is successfull and device is booting fine.

Tracked-On: OAM-118066
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 .../com/android/car/user/CarUserService.java  | 20 +++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/service/src/com/android/car/user/CarUserService.java b/service/src/com/android/car/user/CarUserService.java
index 8958df7f5b..8828dae7de 100644
--- a/service/src/com/android/car/user/CarUserService.java
+++ b/service/src/com/android/car/user/CarUserService.java
@@ -675,11 +675,13 @@ public final class CarUserService extends ICarUserService.Stub implements CarSer
     }
 
     private void stopAllPassenger() {
-        if (!mRunningPassengersId.isEmpty()) {
-            for (int i = 0; i < mRunningPassengersId.size(); i++) {
-                stopPassengerInternal(mRunningPassengersId.get(i), true);
+        synchronized (mLockUser) {
+            if (!mRunningPassengersId.isEmpty()) {
+                for (int i = 0; i < mRunningPassengersId.size(); i++) {
+                    stopPassengerInternal(mRunningPassengersId.get(i), true);
+                }
+                mRunningPassengersId.clear();
             }
-            mRunningPassengersId.clear();
         }
     }
 
@@ -2385,11 +2387,13 @@ public final class CarUserService extends ICarUserService.Stub implements CarSer
 
         mInitialUserSetter.setLastActiveUser(toUserId);
 
-        if (!mRunningPassengersId.isEmpty()) {
-            for (int i = 0; i < mRunningPassengersId.size(); i++) {
-                stopPassengerInternal(mRunningPassengersId.get(i), false);
+        synchronized (mLockUser) {
+            if (!mRunningPassengersId.isEmpty()) {
+                for (int i = 0; i < mRunningPassengersId.size(); i++) {
+                    stopPassengerInternal(mRunningPassengersId.get(i), false);
+                }
+                mRunningPassengersId.clear();
             }
-            mRunningPassengersId.clear();
         }
 
         if (mEnablePassengerSupport) {
-- 
2.34.1

